Yoast/wordpress-seo
# Make url preview in snippet editor reflects permalink consistently

## Context


* When a user uses a custom permalink structure that includes `category`, in Block editor, any changes made to the category will not be reflected in the Url preview inside the snippet editor upon saving.
   * The changes are only observed when the user click save, leave the post and visit the post again. 
* When a user has a nested page, the parent page will not be shown in the Url preview inside the snippet editor, while the permalink previews the parent page as well.
* When a user has a custom post type, the name of the custom post type is not displayed in the Url preview inside the snippet editor, while the permalink previews the custom post type as well.
* This PR is to address the problems above.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the url preview inside snippet editor doesn't reflect the permalink consistently.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

Run testing instructions in two environments below and in both Default and Classic editor (**reason**: during the initial investigation, we had different results for the same situation in both envs. Testing in both envs should be done in order to be sure that the solution works)
1. By building the plugin in Docker
2. By creating a zip from the branch and using it for testing
   * To create a zip run this commands below:
      *   `composer install && yarn install --check-files && grunt build:dev`
      *   `grunt artifact`

* **Test Custom post types**
   * Using CPT UI plugin, create a custom post
   * Confirm that the url preview inside snippet editor reflects the in permalink
   
* **Test custom permalink with categories**
   * Go to Settings > Permalinks > Common Settings and use category in your custom permalink structure, e.g.  
   <img width="905" alt="Screenshot 2022-03-16 at 16 31 39" src="https://user-images.githubusercontent.com/48715883/158627794-41db48ab-02eb-4fb3-87ce-d299de25eb5e.png">
   
   * Create a regular post/open an existing one
   * Check one category for your post and save the post
   * Confirm that the category is also shown in the url preview and in permalink
   * Uncheck the category and check a different one
   * Save your changes
   * Confirm that the changes are reflected in the url preview in the snippet editor and in permalink preview 

* **Test nested pages**
   * Create multiple pages
   * Open a page and assign one of the pages as the parent
   * Confirm that the title of the parent page is also shown in url preview and in permalink preview
   * Assign a different parent page and save the changes
   * Confirm that the changes are reflected in the url preview and in permalink preview

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes #
